druid/src/scroll_component.rs

```diff
@@ -421,65 +385,57 @@ impl<T: Data, W: Widget<T>> Widget<T> for Scroll<T, W> {
         }
     }
 
-    fn lifecycle(&mut self, ctx: &mut LifeCycleCtx, event: &LifeCycle, data: &T, env: &Env) {
+    /// Checks if the lifecycle event applies to the scroll behavior, uses it and returns true if so
+    ///
+    /// Returns false if the lifecycle event was not used
+    pub fn filter_lifecycle(
+        &mut self,
+        ctx: &mut LifeCycleCtx,
+        event: &LifeCycle,
+        env: &Env,
+    ) -> bool {
         match event {
             LifeCycle::AnimFrame(interval) => {
                 // Guard by the timer id being invalid, otherwise the scroll bars would fade
-                // immediately if some other widgeet started animating.
+                // immediately if some other widget started animating.
                 if self.scrollbars.timer_id == TimerToken::INVALID {
```

Comment:
This sounds... wrong? We should have an id for the timer, and ignore ids that aren't ours, surely? I'm not sure why we would use INVALID for all scrolls? (No need to address this in this PR if you don't have an answer - it's fairly unrelated)

Reply:
That does indeed look wrong. This specific code is untouched from `Scroll` so I'd like to look more into it after this PR is merged and have a more focused discussion on it as part of the goal of this PR is to be functionally identical. I've spotted at least one other bug in testing which was around prior to this patch which is persisted and I plan on looking into afterwards.

Reply:
So after closer inspection I am fairly confident in saying that this is not wrong in how it uses `TimerToken::INVALID`, if someone more knowledgeable disagrees please do correct me. Furthermore lifecycle events are no longer filtered so this is not a concern anymore